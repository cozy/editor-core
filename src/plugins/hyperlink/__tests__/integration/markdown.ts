import { BrowserTestCase } from '@atlaskit/webdriver-runner/runner';
import Page from '@atlaskit/webdriver-runner/wd-wrapper';

import {
  goToEditorTestingWDExample,
  mountEditor,
} from '../../../../__tests__/__helpers/testing-example-helpers';
import {
  fullpage,
  editable,
  getDocFromElement,
} from '../../../../__tests__/integration/_helpers';
import { hyperlinkSelectors } from '../../../../__tests__/__helpers/page-objects/_hyperlink';
import { copyHyperlink } from '../../../../__tests__/__helpers/page-objects/_hyperlink';

BrowserTestCase(
  'can insert hyperlink via markdown',
  {},
  async (client: any, testName: string) => {
    const page = await goToEditorTestingWDExample(client);
    await mountEditor(page, {
      appearance: fullpage.appearance,
    });

    await page.type(editable, '[Atlassian](http://atlassian.com)');
    await page.waitFor(hyperlinkSelectors.hyperlink);

    const doc = await page.$eval(editable, getDocFromElement);
    expect(doc).toMatchCustomDocSnapshot(testName);
  },
);

BrowserTestCase(
  'can insert hyperlink via markdown when pasting URL',
  {},
  async (client: any, testName: string) => {
    const page = new Page(client);
    await copyHyperlink(page, 'http://atlassian.com');

    await goToEditorTestingWDExample(client);
    await mountEditor(page, {
      appearance: fullpage.appearance,
    });

    await page.type(editable, '[Atlassian](');
    await page.paste();
    await page.type(editable, ')');
    await page.waitFor(hyperlinkSelectors.hyperlink);

    const doc = await page.$eval(editable, getDocFromElement);
    expect(doc).toMatchCustomDocSnapshot(testName);
  },
);
