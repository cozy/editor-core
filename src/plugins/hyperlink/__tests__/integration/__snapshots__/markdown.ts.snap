// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can insert hyperlink via markdown 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "marks": Array [
            Object {
              "attrs": Object {
                "__confluenceMetadata": null,
                "href": "http://atlassian.com",
              },
              "type": "link",
            },
          ],
          "text": "Atlassian",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`can insert hyperlink via markdown when pasting URL 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "marks": Array [
            Object {
              "attrs": Object {
                "__confluenceMetadata": null,
                "href": "http://atlassian.com",
              },
              "type": "link",
            },
          ],
          "text": "Atlassian",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;
